File: javascript/linkedlist/Node.ts
class Node {
    private data: Number | null;
    private next: Node | null;
    constructor(d: Number, n: Node | null) {
        this.data = d;
        this.next = n;
    }

    getData() {
        return this.data;
    }
    getNext() {
        return this.next;
    }
    setNext(n: Node | null) {
        this.next = n;
    }

}

export default Node;

File: Java/AbstractLinkedList/src/Node.java
public class Node {
    int data;
    Node next;
    Node previous;

    Node(int d, Node n, Node p) {
        this.data = d;
        this.next = n;
        this.previous = p;
    }

    public void setData(int d) {
        this.data = d;
    }

    public int getData() {
        return this.data;
    }

    public void setNext(Node n) {
        this.next = n;
    }

    public Node getNext() {
        return this.next;
    }

    public void setPrevious(Node n) {
        this.previous = n;
    }

    public Node getPrevious() {
        return this.previous;
    }
}
